apps/studio/components/interfaces/Auth/ThirdPartyAuthForm/CreateAuth0Dialog.tsx

```diff
@@ -0,0 +1,182 @@
+import { zodResolver } from '@hookform/resolvers/zod'
+import { Trash } from 'lucide-react'
+import { useEffect } from 'react'
+import { SubmitHandler, useForm } from 'react-hook-form'
+import toast from 'react-hot-toast'
+import * as z from 'zod'
+
+import { useParams } from 'common'
+import { useCreateThirdPartyAuthIntegrationMutation } from 'data/third-party-auth/integration-create-mutation'
+import {
+  Button,
+  Dialog,
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+  DialogSection,
+  DialogTitle,
+  FormControl_Shadcn_,
+  FormField_Shadcn_,
+  Form_Shadcn_,
+  Input_Shadcn_,
+  Separator,
+} from 'ui'
+import { FormItemLayout } from 'ui-patterns/form/FormItemLayout/FormItemLayout'
+
+interface CreateAuth0IntegrationProps {
+  visible: boolean
+  onClose: () => void
+  // TODO: Remove this if this Dialog is only used for creating.
+  onDelete: () => void
+}
+
+const FORM_ID = 'create-auth0-auth-integration-form'
+
+const FormSchema = z.object({
+  enabled: z.boolean(),
+  auth0DomainName: z
+    .string()
+    .trim()
+    .min(1)
+    .regex(/^[A-Za-z0-9-.]+$/, 'The project ID contains invalid characters.'), // Only allow alphanumeric characters and hyphens.
+})
+
+export const CreateAuth0IntegrationDialog = ({
+  visible,
+  onClose,
+  onDelete,
+}: CreateAuth0IntegrationProps) => {
+  // TODO: Remove this if this Dialog is only used for creating.
+  const isCreating = true
+
+  const { ref: projectRef } = useParams()
+  const { mutate: createAuthIntegration, isLoading } = useCreateThirdPartyAuthIntegrationMutation({
+    onSuccess: () => {
+      toast.success(`Successfully created a new Auth0 Auth integration.`)
```

Comment:
there seems to be a very long delay between the toast success and the UI actually showing the newly created Auth0 integration - unless i might be missing something?

when the toast success shows and the modal closes, i still see no integrations for quite a while

Reply:
hmm odd this happened the first time i created an integration, but subsequently it seems okay now

Reply:
yeah i can't seem to repro this anymore - feel free to ignore